### PR TITLE
Updates to metadata on publish sent to legacy

### PR DESCRIPTION
**Pull request by hannako**
This PR is part of [this trello card](https://trello.com/c/7XtWXb4p/3-sync-changes-to-existing-datasets-on-publish-to-legacy)

There are three requirements:

- updates to dataset metadata are sent to legacy 
- updates to datafiles are sent to legacy
- creation of new datasets and datafiles are sent to legacy

This PR implements the first requirement. Comments v welcome on how best to test this @govuklaurence ! 

Have also made the users created by seeds.rb belong to real organisations so that their datasets populate the manage_datasets page

### CODE REVIEW

File: app/workers/legacy_sync_worker.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+require 'rest-client'
+
+class LegacySyncWorker
+  include Sidekiq::Worker
+
+  def perform(id)
+    json = Dataset.find(id).ckanify_metadata.to_json
+    server =
+      Rails.env.production? ? "https://data.gov.uk" : "https://test.data.gov.uk"
```
Comment (hannako) [inline app/workers/legacy_sync_worker.rb]: The worker is only called if the env is production, but have added as a second line of defence as I was nervous of breaking legacy! Perhaps there's a better safety mechanism...

File: app/models/dataset.rb [hunk 2]
```diff
@@ -78,6 +78,46 @@ def as_indexed_json(_options={})
     )
   end
 
+  # map the dataset.as_json so that the keys are in a format that ckan recognises
+  def ckanify_metadata
```
Comment (TobyRet) [inline app/models/dataset.rb]: Are we 100% sure all those fields will be present?

File: spec/models/dataset_spec.rb [hunk 3]
```diff
@@ -95,4 +95,35 @@
 
     expect(Dataset.count).to eq 0
   end
+
+  it "can ckanify its metadata" do
```
Comment (TobyRet) [inline spec/models/dataset_spec.rb]: Related to the comment above. This tests the 'happy' path but do we need to think of edge cases?

File: spec/workers/legacy_sync_worker.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+require 'rails_helper'
+
```
Comment (TobyRet) [inline spec/workers/legacy_sync_worker.rb]: I see there is a testing library for sidekiq. https://github.com/mperham/sidekiq/wiki/Testing
Would be good to hear from the others whether they feel its worth adding tests for all workers.

File: app/workers/legacy_sync_worker.rb (hunk 1 above)
Comment (maxf) [reply app/workers/legacy_sync_worker.rb]: This means running tests requires test.data.gov.uk? That's probably not ideal as it depends on test.dgu being up, but also we cannot run the tests offline. Is mocking a possibility?

Comment (laurentqro) [reply app/workers/legacy_sync_worker.rb]: yeah I think it might be a good idea to mock data.gov.uk in our tests here.

File: spec/workers/legacy_sync_worker.rb (hunk 4 above)
Comment (govuklaurence) [reply spec/workers/legacy_sync_worker.rb]: It would be helpful, but as long as the lib code works it should be good to go. Workers should really only ever call code inside the app

File: db/seeds.rb [hunk 5]
```diff
@@ -33,15 +33,9 @@
...
-land_registry.title = "Land Registry"
-land_registry.save!()
-
-hmrc = Organisation.new
-hmrc.name = "hmrc"
-hmrc.title = "hmrc"
-hmrc.save!()
+land_registry = Organisation.find_by(name: 'land-registry')
+hmrc = Organisation.find_by(name: 'hmrc')
+gds = Organisation.find_by(name: 'government-digital-service')
```
Comment (laurentqro) [inline db/seeds.rb]: This will not find anything if it's the first time I run rake db:seed.

File: spec/workers/legacy_sync_worker.rb (hunk 4 above)
Comment (laurentqro) [reply spec/workers/legacy_sync_worker.rb]: yeah, I'd test the logic that the worker runs, rather than test the worker itself, if that makes sense.

File: app/workers/legacy_sync_worker.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    url = "#{ server }/api/3/action/package_patch"
+    headers = {Authorization: "6c25593f-19fe-4064-a451-ddc0aff562d9"}
```
Comment (govuklaurence) [inline app/workers/legacy_sync_worker.rb]: this needs to be in an env var, since it's a secret

File: setup.sh [hunk 7]
```diff
@@ -14,7 +14,7 @@ print_error() {
 
 export SECRET_KEY_BASE="foobar"
 bundle install
-rails db:reset db:migrate db:seed
+rails db:drop db:create db:schema:load
```
Comment (govuklaurence) [inline setup.sh]: still needs to `db:seed`

Comment (hannako) [reply setup.sh]: 👍
